File: src/Page/SelectedCoursesPage.java
package Page;

import SystemMessage.FunctionType;
import SystemMessage.SystemMessage;

public class SelectedCoursesPage extends Page{
	
	private int numberOfDropableCourses;
	
	public SelectedCoursesPage(String content) {
		super(content);
		setType(PageType.SELECTED_COURSES_PAGE);
		setName("Selected Courses Page");
		
	}
	
	@Override
	public SystemMessage runPage() {
		showContent();
		String input;
		while(true) {
			input = takeInput();
			try {
				int selection = Integer.parseInt(input);
				if (selection > numberOfDropableCourses || selection < 0) {
					System.out.println("Enter a valid number");
				}
				else {
					return new SystemMessage(FunctionType.DROP_COURSE, PageType.SELECTED_COURSES_PAGE, selection);
				}
			}
			catch (NumberFormatException e) {
				if (input.equalsIgnoreCase("a")) {
					return new SystemMessage(FunctionType.SEND_APPROVE, PageType.MAIN_MENU_PAGE, null);
				}
				else if (input.equalsIgnoreCase("q")) {
					return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.MAIN_MENU_PAGE, null);
				}
				else {
					System.out.println("Wrong Input");
				}
			}
		}		
	}
	
	/////////// GETTER - SETTER METHODS ////////////
	public int getNumberOfDropableCourses() {
		return numberOfDropableCourses;
	}

	public void setNumberOfDropableCourses(int numberOfDropableCourses) {
		this.numberOfDropableCourses = numberOfDropableCourses;
	}
}

File: python_development/Page/SelectedCoursesPage.py
from python_development.Page.Page import Page
from python_development.Page.PageType import PageType
from python_development.System.FunctionType import FunctionType
from python_development.System.SystemMessage import SystemMessage

class SelectedCoursesPage(Page):

    numberOfDropableCourses = None

    def __init__(self, content):
        super().__init__(content)
        self.setType("SELECTED_COURSES_PAGE")
        self.setName("Selected Courses Page")


    def runPage(self):
        self.showContent()
        inputFromPage = None
        while(True):
            inputFromPage = self.takeInput()
            try:
                selection = int(inputFromPage)
                if (selection > self.numberOfDropableCourses or selection < 0):
                    print("Enter a valid number")
                else:
                    return SystemMessage(FunctionType.DROP_COURSE, PageType.SELECTED_COURSES_PAGE, selection)
            except:
                if(inputFromPage.lower() == "a"):
                    return SystemMessage(FunctionType.SEND_APPROVE, PageType.MAIN_MENU_PAGE, None)
                if(inputFromPage.lower() == "q"):
                    return SystemMessage(FunctionType.CHANGE_PAGE, PageType.MAIN_MENU_PAGE, None)
                else:
                    print("Wrong Input")


    def getNumberOfDropableCourses(self):
        return self.numberOfDropableCourses
    
    def setNumberOfDropableCourses(self, numberOfDropableCourses):
        self.numberOfDropableCourses = numberOfDropableCourses
        